default/AI/ShipDesignAI.py

```diff
@@ -790,6 +816,102 @@ def update_stats(self, ignore_species=False):
             if self.attacks:
                 self.attacks = foAI.foAIstate.weight_attacks(self.attacks, weapons_grade)
 
+    def _apply_hardcoded_effects(self):
+        """Update stats that can not be read out by the AI yet, i.e. applied by effects.
+
+        This function should contain *all* hardcoded effects for hulls/parts to be considered by the AI
+        to make sure we can easily adjust the values for future balance changes or after implementing a
+        method to read out all stats.
+        """
+        REGULAR_HULL_DETECTION = 25
+
+        def _irregular_detection(detection):
+            self.detection -= REGULAR_HULL_DETECTION
+            self.detection += detection
+
+        def _organic_growth(growth_per_turn, maximum):
+            self.organic_growth = growth_per_turn
+            self.maximum_organic_growth = maximum
+
+        hullname = self.hull.name
+        self.detection += REGULAR_HULL_DETECTION  # overwrite if special hull
+        if hullname == "SH_ROBOTIC":
+            self.repair_per_turn += 2
+        elif hullname == "SH_SPATIAL_FLUX":
+            self.stealth = max(0, self.stealth-30)
+        elif hullname == "SH_NANOROBOTIC":
+            self.repair_per_turn = self.structure
+        elif hullname == "SH_LOGISTICS_FACILITATOR":
+            self.repair_per_turn = self.structure  # TODO: Add modifier for fleet repair
```

Comment:
Actually, Morlic, could you please move these hardcoded values over to AIDependencies.py, and Mat, then that would be the place to update things.  I want to try to make it so that anytime a non-AI dev changes something and wants to check if there is a known adjustment for them to make to the AI, they only need to check AIDependencies.py and not have to search all over the AI code set.